### DISCUSSION TITLE
asreview loading time

### DISCUSSION THREAD

**Discussion by DBergman1997** [asreview/asreview#797] [Q&A]
I downloaded asreview version 0.18. 

When I selected my prior knowledge and the options:

**Classifier:** Logistic regression
**Query strategy:** Maximum
**Feature extraction:** Doc2Vec

Now asreview is loading for around 3,5 hours, is this normal or is something wrong? 

Thanks in advance!

**Comment by jteijema** [comment]
Hi @DBergman1997 ,

Can you share with us how big your dataset is? That way I can better judge if this is normal behaviour.

**Reply by jteijema** [reply]
Because Doc2Vec is a shallow neural network, training it can take a long time. However, 3.5 hours is very long indeed and there might be something else going on. 

Does the console report report any issues?

**Reply by Rensvandeschoot** [reply]
We recently conducted a simulation study on a dataset with 50K labels, and the different training times are in this table. @pablov-1995 on what type of computer did you run these analyses??

![image](https://user-images.githubusercontent.com/36502709/137308271-6a3c3698-6a11-4984-84b1-68235051645b.png)

**Reply by DBergman1997** [reply]
> Hi @DBergman1997 ,
> 
> Can you share with us how big your dataset is? That way I can better judge if this is normal behaviour.

18600!

**Reply by jteijema** [reply]
18600! = 10^71337.59724740479

All kidding aside, judging from the table @Rensvandeschoot provided, the loading phase of ASReview should not be taking that long for <20.000 documents. I know that @pablov-1995 did not use a very fast processing platform.

Did the loading finish by now? If not, I'd like to try for myself with the same dataset and see if I can replicate the results. If you were able to start, I will do some testruns on my own at a later time to see what loading times are considered within margins.

Let me know!

**Reply by pablov-1995** [reply]
Hi @Rensvandeschoot @JTeijema! I used an HP Omen laptop with the following specs for the simulations:

CPU: AMD Ryzen 7-4800H (8 cores / 16 threads) 
RAM: 16GB
GPU: Nvidia RTX2060-6GB

**Reply by jteijema** [reply]
So, for a laptop very fast, but not a lot compared to a dedicated computation server. I will run the dataset myself to see if anything slows down the process.

**Reply by JonnaLind** [reply]
Hi there (@jteijema & Co), 

I also have really long loading times for my 33K dataset (including 330 items with prior knowledge). It's running on a server, that I don't know the specifications of, but I've been loading datasets of a few thousands quite quickly earlier. I used to use ris-files, now I'm attempting a big (26 mb) excel file. Earlier, a small (330 items) excel file gave us no trouble. 

I've read here, that 50K should last for about 1,5 minute, so the hours I've now been waiting is irregular...  

the model is TF-IDF, Naive Bayes, Mixed (95% maximum and 5% random - also tried with maximum), Dynamic resampling (double). 
[voorbeeld.xlsx](https://github.com/user-attachments/files/18494513/voorbeeld.xlsx)

I've attatched an example of the structure of my file. 

I'd be very happy, if you could help me out!

Kind regards, Jonna

**Reply by JonnaLind** [reply]
I tried sorting the Excel File differently, so the labelled data are in the first records. Only, now I get an error message saying "Oeps, something went wrong"...
Now, I'm trying with a RIS-file, and again getting the error message when trying to import, this time with a "Whoops". 
I also tried with a RIS-file containing 17K items, but it was whoopsed already at import. When importing 14K ris-file, I got a short Whoops, but the machine composed itsef and went on to having imported 14K records in RIS. Trying to warm up the AI (default settings) after less than 1 minute resulted in a Whoops. And a "503 Service Temporarily Unavailable" after refresh (twice). I now have to restart. Might be an issue with our hosting.... (we're now trying that avenue...)

**Reply by J535D165** [reply]
Can reach out via asreview@uu.nl? It might be something specific to your organization.

**Reply by JonnaLind** [reply]
Thanks! we're looking at it together with our hosting partner. i might delete this later, of add some info from the hosting partner.
